x-pack/plugins/enterprise_search/server/utils/search_result_provider.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { from, takeUntil } from 'rxjs';
+
+import { IBasePath } from '@kbn/core-http-server';
+import { GlobalSearchResultProvider } from '@kbn/global-search-plugin/server';
+
+import { CONNECTOR_DEFINITIONS } from '../../common/connectors/connectors';
+import {
+  ENTERPRISE_SEARCH_CONNECTOR_CRAWLER_SERVICE_TYPE,
+  ENTERPRISE_SEARCH_CONTENT_PLUGIN,
+} from '../../common/constants';
+
+export function toSearchResult({
+  basePath,
+  iconPath,
+  name,
+  score,
+  serviceType,
+}: {
+  basePath: IBasePath;
+  iconPath: string;
+  name: string;
+  score: number;
+  serviceType: string;
+}) {
+  return {
+    icon: iconPath
+      ? basePath.prepend(`/plugins/enterpriseSearch/assets/source_icons/${iconPath}`)
+      : 'logoEnterpriseSearch',
+    id: serviceType,
+    score,
+    title: name,
+    type: 'Enterprise Search',
+    url: {
+      path: `${ENTERPRISE_SEARCH_CONTENT_PLUGIN.URL}/search_indices/new_index/${
+        serviceType === ENTERPRISE_SEARCH_CONNECTOR_CRAWLER_SERVICE_TYPE
+          ? 'crawler'
+          : `connector?service_type=${serviceType}`
+      }`,
+      prependBasePath: true,
+    },
+  };
+}
+
+export function getSearchResultProvider(basePath: IBasePath): GlobalSearchResultProvider {
```

Comment:
Should we pass the `Config` to this so we can exclude things that are gated? Thinking about for serverless we'd want to honor the `hasWebCrawler` config option for example.

Reply:
That's a good point, let me fix that.

Reply:
Done in latest